Source/DafnyCore/Dafny.atg

```diff
@@ -2072,15 +2073,17 @@ InvariantClause<. List<AttributedExpression> invariants.> =
   .
 
 /*------------------------------------------------------------------------*/
-MethodSpec<.bool isGhost, List<AttributedExpression> req, List<FrameExpression> mod, List<AttributedExpression> ens,
+MethodSpec<.bool isGhost, List<AttributedExpression> req, List<FrameExpression> reads, List<FrameExpression> mod, List<AttributedExpression> ens,
             List<Expression> decreases, ref Attributes decAttrs, ref Attributes modAttrs, string caption, bool performThisDeprecatedCheck.>
 = (. Contract.Requires(cce.NonNullElements(req));
+     Contract.Requires(cce.NonNullElements(reads));
      Contract.Requires(cce.NonNullElements(mod));
      Contract.Requires(cce.NonNullElements(ens));
      Contract.Requires(cce.NonNullElements(decreases));
   .)
   SYNC
-  { ModifiesClause<ref mod, ref modAttrs, false, performThisDeprecatedCheck>
+  { ReadsClause<reads, false, false, true>
```

Comment:
It feeks weird that you parse reads clauses on method even if the flag --reads-clauses-on-method is set to false. Since we now have project files, I think you should at least emit a warning if you parse reads clauses but they are not verified.

Reply:
They are already a hard error, see `ReadsOnMethods-Resolution.dfy`